### PR TITLE
Refactors throughout the app

### PR DESCRIPTION
**Pull request by victorbalan**
- Remove deprecated `output-test-console.html`
- Refactor `test-stats-console.html`' (fixed loading and split the init function)
- Add challenge-service, submission-service, task-service
- Extract time formater in util
- Add robot lang + new icons for the robot task
- Add user-task-detail
- Fetch real skills for the user challenge

### CODE REVIEW

File: app/scripts/util.js [hunk 1]
```diff
@@ -85,6 +85,12 @@ var util = {
 		}
 		page.redirect('/error');
 	},
+	getFormatedTime: function(d){
+		var h = Math.floor(d / 3600) % 24;
+		var m = Math.floor(d / 60 % 60);
+		var s = Math.floor(d % 60);
+		return ('0' + h).slice(-2) + ':' +	('0' + m).slice(-2) + ':' + ('0' + s).slice(-2);
```
Comment (lorenzleutgeb) [inline app/scripts/util.js]: Random tab.

File: app/elements/code-editor/code-editor.html [hunk 2]
```diff
@@ -66,26 +61,40 @@
...
+						mode = 'ace/mode/c_cpp';
 						break;
 				}
-				this.fileName = fileName;
 				this.$.editor.getSession().setMode({
 					path: mode,
 					v: Date.now()
 				});
+			},
+			_getFile: function(lang){
```
Comment (flowlo) [inline app/elements/code-editor/code-editor.html]: Ideally we should not need this, and rather use the filename we get from the templates.

File: app/elements/user/challenge/user-task-detail.html [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			},
+			_submissionsChanged: function(submissions){
+				if(submissions.length===0){
+						this.noSubmissions = true;
+						this.code = 'No submissions were made for this task.';
+						return;
+				}
+				this.noSubmissions = false;
+				var element = this;
+				// TODO: Fuck my life, javascript and polymer. Burn in hell!
```
Comment (lorenzleutgeb) [inline app/elements/user/challenge/user-task-detail.html]: d

Comment (victorbalan) [inline app/elements/user/challenge/user-task-detail.html]: please let's leave this here to know what we're dealing with.

Comment (lorenzleutgeb) [inline app/elements/user/challenge/user-task-detail.html]: You can add a TODO, but not swear.

**Pull request merged**